### ISSUE TITLE
lxc-usernet config seem to use all veth bound to a bridge when computing one user's quota.

### ISSUE THREAD

**Issue by P-EB** [lxc/lxc#2765]
Hi,

Debian maintainer of lxc3 here. I've met an issue with lxc.

 * Distribution: Debian
 * Distribution version: 10/buster
 * The output of
   * `lxc-start --version` 3.1.0
   * `lxc-checkconfig`

```
Kernel configuration not found at /proc/config.gz; searching...                                                                                                                                                   
Kernel configuration found at /boot/config-4.18.0-2-amd64                                                                                                                                                         
--- Namespaces ---                                                                                                                                                                                                
Namespaces: enabled                                                                                                                                                                                               
Utsname namespace: enabled                                                                                                                                                                                        
Ipc namespace: enabled                                                                                                                                                                                            
Pid namespace: enabled                                                                                                                                                                                            
User namespace: enabled                                                                                                                                                                                           
Network namespace: enabled                                                                                                                                                                                        

--- Control groups ---
Cgroups: enabled

Cgroup v1 mount points:
/sys/fs/cgroup/systemd
/sys/fs/cgroup/net_cls,net_prio
/sys/fs/cgroup/blkio
/sys/fs/cgroup/cpu,cpuacct
/sys/fs/cgroup/cpuset
/sys/fs/cgroup/memory
/sys/fs/cgroup/devices
/sys/fs/cgroup/freezer
/sys/fs/cgroup/perf_event
/sys/fs/cgroup/pids

Cgroup v2 mount points:
/sys/fs/cgroup/unified

Cgroup v1 clone_children flag: enabled
Cgroup device: enabled
Cgroup sched: enabled
Cgroup cpu account: enabled
Cgroup memory controller: enabled
Cgroup cpuset: enabled

--- Misc ---
Veth pair device: enabled, loaded
Macvlan: enabled, not loaded
Vlan: enabled, not loaded
Bridges: enabled, loaded
Advanced netfilter: enabled, loaded
CONFIG_NF_NAT_IPV4: enabled, loaded
CONFIG_NF_NAT_IPV6: enabled, not loaded
CONFIG_IP_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_IP6_NF_TARGET_MASQUERADE: enabled, not loaded
CONFIG_NETFILTER_XT_TARGET_CHECKSUM: enabled, not loaded
CONFIG_NETFILTER_XT_MATCH_COMMENT: enabled, not loaded
FUSE (for use with lxcfs): enabled, loaded

--- Checkpoint/Restore ---
checkpoint restore: enabled
CONFIG_FHANDLE: enabled
CONFIG_EVENTFD: enabled
CONFIG_EPOLL: enabled
CONFIG_UNIX_DIAG: enabled
CONFIG_INET_DIAG: enabled
CONFIG_PACKET_DIAG: enabled
CONFIG_NETLINK_DIAG: enabled
File capabilities:

Note : Before booting a new kernel, you can check its configuration
usage : CONFIG=/path/to/config /usr/bin/lxc-checkconfig
```
   * `uname -a` `Linux XXX 4.18.0-2-amd64 #1 SMP Debian 4.18.10-2 (2018-10-07) x86_64 GNU/Linux`
   * `cat /proc/self/cgroup`

```
10:pids:/user.slice/user-1000.slice/session-436.scope
9:perf_event:/
8:freezer:/user/pgsql/0
7:devices:/user.slice
6:memory:/user/pgsql/0
5:cpuset:/
4:cpu,cpuacct:/user.slice
3:blkio:/user.slice
2:net_cls,net_prio:/
1:name=systemd:/user/pgsql/0
0::/user.slice/user-1000.slice/session-436.scope
```
   * `cat /proc/1/mounts`
```
sysfs /sys sysfs rw,nosuid,nodev,noexec,relatime 0 0
proc /proc proc rw,nosuid,nodev,noexec,relatime 0 0
udev /dev devtmpfs rw,nosuid,relatime,size=16369484k,nr_inodes=4092371,mode=755 0 0
devpts /dev/pts devpts rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000 0 0
tmpfs /run tmpfs rw,nosuid,noexec,relatime,size=3277304k,mode=755 0 0
/dev/mapper/pile-slash / ext4 rw,relatime,errors=remount-ro 0 0
securityfs /sys/kernel/security securityfs rw,nosuid,nodev,noexec,relatime 0 0
tmpfs /dev/shm tmpfs rw,nosuid,nodev 0 0
tmpfs /run/lock tmpfs rw,nosuid,nodev,noexec,relatime,size=5120k 0 0
tmpfs /sys/fs/cgroup tmpfs ro,nosuid,nodev,noexec,mode=755 0 0
cgroup2 /sys/fs/cgroup/unified cgroup2 rw,nosuid,nodev,noexec,relatime 0 0
cgroup /sys/fs/cgroup/systemd cgroup rw,nosuid,nodev,noexec,relatime,xattr,name=systemd 0 0
pstore /sys/fs/pstore pstore rw,nosuid,nodev,noexec,relatime 0 0
bpf /sys/fs/bpf bpf rw,nosuid,nodev,noexec,relatime,mode=700 0 0
cgroup /sys/fs/cgroup/net_cls,net_prio cgroup rw,nosuid,nodev,noexec,relatime,net_cls,net_prio 0 0
cgroup /sys/fs/cgroup/blkio cgroup rw,nosuid,nodev,noexec,relatime,blkio 0 0
cgroup /sys/fs/cgroup/cpu,cpuacct cgroup rw,nosuid,nodev,noexec,relatime,cpu,cpuacct 0 0
cgroup /sys/fs/cgroup/cpuset cgroup rw,nosuid,nodev,noexec,relatime,cpuset,clone_children 0 0
cgroup /sys/fs/cgroup/memory cgroup rw,nosuid,nodev,noexec,relatime,memory 0 0
cgroup /sys/fs/cgroup/devices cgroup rw,nosuid,nodev,noexec,relatime,devices 0 0
cgroup /sys/fs/cgroup/freezer cgroup rw,nosuid,nodev,noexec,relatime,freezer 0 0
cgroup /sys/fs/cgroup/perf_event cgroup rw,nosuid,nodev,noexec,relatime,perf_event 0 0
cgroup /sys/fs/cgroup/pids cgroup rw,nosuid,nodev,noexec,relatime,pids 0 0
systemd-1 /proc/sys/fs/binfmt_misc autofs rw,relatime,fd=28,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=1851 0 0
mqueue /dev/mqueue mqueue rw,relatime 0 0
debugfs /sys/kernel/debug debugfs rw,relatime 0 0
hugetlbfs /dev/hugepages hugetlbfs rw,relatime,pagesize=2M 0 0
none /tmp tmpfs rw,relatime 0 0
/dev/md0p1 /boot ext4 rw,relatime 0 0
/dev/mapper/pile-var /var ext4 rw,relatime 0 0
/dev/mapper/pile-home /home ext4 rw,relatime 0 0
/dev/mapper/pile-proxy /home/unpriviledged/revproxy/.local/share/lxc ext4 rw,relatime 0 0
/dev/mapper/pile-gitlab /home/unpriviledged/git/.local/share/lxc ext4 rw,relatime 0 0
fusectl /sys/fs/fuse/connections fusectl rw,relatime 0 0
tmpfs /run/user/1000 tmpfs rw,nosuid,nodev,relatime,size=3277300k,mode=700,uid=1000,gid=1000 0 0
/dev/mapper/pile-mx /home/unpriviledged/mx/.local/share/lxc ext4 rw,relatime 0 0
/dev/mapper/pile-pgsql /home/unpriviledged/pgsql/.local/share/lxc ext4 rw,relatime 0 0
lxcfs /var/lib/lxcfs fuse.lxcfs rw,nosuid,nodev,relatime,user_id=0,group_id=0,allow_other 0 0
```

# Issue description

lxc-usernet quota is supposed to be applied per user, but it's not the case.

# Steps to reproduce

 1. Have all users rely on the same bridge with veth
 2. Turn down all containers
 3. Set all users quota to 1
 4. Try to start one container on one user and one on a second one.

Example: `/etc/lxc/lxc-usernet`
```
revproxy veth lxcbr 1
git veth lxcbr 1
mx veth lxcbr 1
pgsql veth lxcbr 1
```

```
revproxy@xxx $ lxc-start proxy
revproxy@xxx
```

```
git@xxx $ lxc-start gitlab
lxc-start: pgsql: lxccontainer.c: wait_on_daemonized_start: 864 Received container state "ABORTING" instead of "RUNNING"                                                                                          
lxc-start: pgsql: tools/lxc_start.c: main: 330 The container failed to start                                                                                                                                      
lxc-start: pgsql: tools/lxc_start.c: main: 333 To get more details, run the container in foreground mode                                                                                                          
lxc-start: pgsql: tools/lxc_start.c: main: 336 Additional information can be obtained by setting the --logfile and --logpriority options

# hack lxc-usernet and set quotas to 2.
git@xxx $ lxc-start gitlab
git@xxx
```

Of course I used the advice and used logfile + logpriority to TRACE the file is unequivocal:
```
lxc-start proxy 20190101180939.135 INFO     network - network.c:lxc_create_network_unpriv_exec:2150 - Execing lxc-user-nic create /home/unpriviledged/git/.local/share/lxc proxy 6481 veth lxcbr (null)
lxc-start proxy 20190101180939.140 ERROR    network - network.c:lxc_create_network_unpriv_exec:2178 - lxc-user-nic failed to configure requested network: cmd/lxc_user_nic.c: 1296: main: Quota reached
```

It's reproducible (mx can't start unless his quota is raised to 3, etc etc).

I guess it's due to the fact that the quotas are computed against a user, but considering how many ifaces are bound to the bridge (lxcbr here).

It's not *that* dramatic, but the manpage reads

```
       user   is the username to whom this entry applies.

       number is the number or quota of network interfaces of the given type which the given user or group may attach to the given bridge, for instance 2.
```

I can provide more intel if needed.

**Comment by P-EB** [comment]
@brauner a friend of mine under 2.0.7 doesn't have this issue.

Hope this helps.

**Comment by hallyn** [comment]
Sorry, I'm not following.  what does

```
# hack lxc-usernet and set quotas to 2.
```

mean?

**Comment by P-EB** [comment]
@hallyn it means that I replace "1" at the end of each line by "2".

I could have changed it only for the git user, but then at the next stop of another container I couldn't start it.

**Comment by CameronNemo** [comment]
How is this functionality possible without a database of users and assigned interfaces?

**Comment by brauner** [comment]
This should be fixed in #2765.
